docs/authorization/scopes_and_consents/scopes.rst

```diff
@@ -0,0 +1,246 @@
+    ...
+
+    # or, potentially, after there is a concrete client
+    _tc = globus_sdk.TransferClient()
+    client.oauth2_start_flow(requested_scopes=[_tc.scopes.all])
+
+As Instance Attributes and Methods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some client classes only provide their scopes for instances. These cases cover
+services which are distributed or contain multiple subservices with their own
+scopes.
+
+For example, ``GCSClient`` and ``SpecificFlowClient`` each have a ``scopes``
+attribute of ``None`` on their classes.
+
+In the case of ``SpecificFlowClient``, scopes are populated whenever an
+instance is instantiated. So the following usage is valid:
+
+.. code-block:: python
+
+    import globus_sdk
+
+    FLOW_ID = "<YOUR_ID_HERE>"
+
+    client = globus_sdk.SpecificFlowClient(FLOW_ID)
+    flow_user_scope = client.scopes.user
+
+In the case of GCS, a distributed service, ``scopes`` is always ``None``.
+However, :meth:`globus_sdk.GCSClient.get_gcs_endpoint_scopes` and
+:meth:`globus_sdk.GCSClient.get_gcs_collection_scopes` are available helpers
+for getting specific collections of scopes.
+
+Using a Scope Builder to Get Matching Tokens
+--------------------------------------------
+
+A ``ScopeBuilder`` contains the resource server name used to get token data
+from a token response.
+To elaborate on the above example:
+
+.. code-block:: python
+
+    import globus_sdk
+    from globus_sdk.scopes import TransferScopes
+
+    CLIENT_ID = "<YOUR_ID_HERE>"
+
+    client = globus_sdk.NativeAppAuthClient(CLIENT_ID)
+    client.oauth2_start_flow(requested_scopes=[TransferScopes.all])
+    authorize_url = client.oauth2_get_authorize_url()
+    print("Please go to this URL and login:", authorize_url)
+    auth_code = input("Please enter the code you get after login here: ").strip()
+    token_response = client.oauth2_exchange_code_for_tokens(auth_code)
+
+    # use the `resource_server` of a ScopeBuilder to grab the associated token
+    # data from the response
+    tokendata = token_response.by_resource_server[TransferScopes.resource_server]
+
+Scope objects
+-------------
+
+The SDK provides a ``Scope`` object which is the class model for a scope.
+``Scope``\s can be parsed from strings and serialized to strings, and support
+programmatic manipulations to describe dependent scopes.
+
+``Scope`` can be constructed using its initializer, or one of its two main
+parsing methods: ``Scope.parse`` and ``Scope.deserialize``.
+``parse`` produces a list of scopes from a string, while ``deserialize``
+produces exactly one.
+
+For example, one can create a ``Scope`` from the Groups "all" scope
+as follows:
+
+.. code-block:: python
+
+    from globus_sdk.scopes import GroupsScopes, Scope
+
+    group_scope = Scope.deserialize(GroupsScopes.all)
+
+``Scope`` objects primarily provide three main pieces of functionality:
+
+    * parsing (deserializing)
+    * stringifying (serializing)
+    * dynamic scope tree building
```

Comment:
I think this helps to build _any_ scope, right?

```suggestion
    * scope tree construction
```

Reply:
Yes, after a manner. The tree of scopes is primarily meaningful and useful vis-a-vis dynamic dependencies. You can construct a scope tree which mirrors static dependencies, but I'm not sure that doing so serves much purpose.

Reply:
It's possible, but I don't think that the `Scope` object we provide here matches what's provided via the Auth scope management API.

Scopes have underlying IDs, descriptions, and other attributes which we aren't modeling here.
That said, I think this is a nicer API for definition, and it would be interesting to look at it as a way of defining / declaring our static dependency data.